src/java.base/unix/classes/sun/nio/fs/UnixPath.java

```diff
@@ -113,45 +107,18 @@ private static String normalize(String input, int len, int off) {
         return sb.toString();
     }
 
+    private static final jdk.internal.access.JavaLangAccess JLA =
+            jdk.internal.access.SharedSecrets.getJavaLangAccess();
+
     // encodes the given path-string into a sequence of bytes
     private static byte[] encode(UnixFileSystem fs, String input) {
-        SoftReference<CharsetEncoder> ref = encoder.get();
-        CharsetEncoder ce = (ref != null) ? ref.get() : null;
-        if (ce == null) {
-            ce = Util.jnuEncoding().newEncoder()
-                .onMalformedInput(CodingErrorAction.REPORT)
-                .onUnmappableCharacter(CodingErrorAction.REPORT);
-            encoder.set(new SoftReference<>(ce));
-        }
-
-        char[] ca = fs.normalizeNativePath(input.toCharArray());
-
-        // size output buffer for worse-case size
-        byte[] ba = new byte[(int)(ca.length * (double)ce.maxBytesPerChar())];
-
-        // encode
-        ByteBuffer bb = ByteBuffer.wrap(ba);
-        CharBuffer cb = CharBuffer.wrap(ca);
-        ce.reset();
-        CoderResult cr = ce.encode(cb, bb, true);
-        boolean error;
-        if (!cr.isUnderflow()) {
-            error = true;
-        } else {
-            cr = ce.flush(bb);
-            error = !cr.isUnderflow();
-        }
-        if (error) {
+        input = fs.normalizeNativePath(input);
+        try {
+            return JLA.getBytesNoRepl(input, Util.jnuEncoding());
+        } catch (CharacterCodingException cce) {
```

Comment:
The encode method pre-dates JLA.getBytesNoRepl and the recent optimisations so this is a good cleanup.